packages/axios-extension/src/abap/ui5-abap-repository-service.ts

```diff
@@ -73,27 +85,33 @@ export class Ui5AbapRepositoryService extends ODataService {
     /**
      * Deploy the given archive either by creating a new BSP or updating an existing one.
      *
-     * @param archivePath path to a zip archive containing the application files
+     * @param archive zip archive containing the application files as buffer
      * @param app application configuration
      * @param testMode if set to true, all requests will be sent, the service checks them, but no actual deployment will happen
+     * @param safeMode if set then the SafeMode url parameter will be set. SafeMode is by default active, to activate provide false
      * @returns the Axios response object for futher processing
      */
-    public async deploy(archivePath: string, app: ApplicationConfig, testMode = false): Promise<AxiosResponse> {
+    public async deploy(
```

Comment:
@ullasholla since I am anyhow doing a breaking change here, should I also switch to a destructured object for the parameters

Reply:
Yes, please

Reply:
https://github.com/SAP/open-ux-tools/pull/797/commits/02a067037dc3e012ae8bcc3b3f817eb9b190c18b